include/simdjson/generic/ondemand/document-inl.h

```diff
@@ -68,7 +68,10 @@ simdjson_inline simdjson_result<object> document::start_or_resume_object() noexc
 simdjson_inline simdjson_result<value> document::get_value() noexcept {
   // Make sure we start any arrays or objects before returning, so that start_root_<object/array>()
   // gets called.
-  iter.assert_at_document_depth();
+#if SIMDJSON_DEVELOPMENT_CHECKS
+  if (!iter.at_root()) { return OUT_OF_ORDER_ITERATION; }
```

Comment:
Could you please clarify why this if under #ifdef, while the check `iter.assert_at_root();` below is always done? If the performance is your concern, wouldn't it worth to put both under #ifdef? If not, maybe move both out of it?

Reply:
It is the convention throughout the code that  the macro `SIMDJSON_DEVELOPMENT_CHECKS` determines whether we check for OUT_OF_ORDER_ITERATION. The logic behind it is that these errors only occurs when the code that was written while breaking some simdjson::ondemand requirement. They should not occur in released code after these issues were fixed.

The `iter.assert_at_root();` is different. In Debug mode, when `SIMDJSON_DEVELOPMENT_CHECKS` is unset, it may not serve a purpose in this case, but when compiling in release mode, it generates SIMDJSON_ASSUME statements which instruct the compilers to make some assumptions. 

I will add comments to the code.

Reply:
Comments added.